static/app/components/globalFeedbackForm.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  createContext,
+  type ReactNode,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+} from 'react';
+import type {FeedbackDialog} from '@sentry/types';
+
+import {
+  type FeedbackIntegration,
+  useFeedback,
+  type UseFeedbackOptions,
+} from 'sentry/components/feedback/widget/useFeedback';
+
+interface GlobalFeedbackFormProps {
+  children: ReactNode;
+}
+
+interface GlobalFeedbackFormContextType {
+  feedback: FeedbackIntegration | undefined;
+  openForm: (options?: UseFeedbackOptions) => Promise<void>;
+}
+
+const GlobalFeedbackFormContext = createContext<GlobalFeedbackFormContextType>({
+  feedback: undefined,
+  openForm: Promise.resolve,
+});
+
+/**
+ * Returns two properties:
+ *
+ * - `feedback`: The feedback integration object. If undefined, the feedback
+ *               integration is not available and feedback buttons should not
+ *               be rendered.
+ * - `openForm`: A function that opens the feedback form. It accepts some
+ *               option overrides to change how the form is displayed.
+ *
+ *  Usage:
+ *
+ *  ```tsx
+ * const {feedback, openForm} = useFeedbackForm();
+ *
+ * if (!feedback) {
+ *   return null;
+ * }
+ *
+ * return <Button onClick={() => openForm()}>{t('Give Feedback')}</button>;
```

Comment:
What if `openForm` was nullable, whenever a form cannot be opened the function itself is unavailable. One less thing to import, and still gives callers the chance to check.


```suggestion
 * - `openForm`: A function that opens the feedback form. It accepts some
 *               option overrides to change how the form is displayed.  If undefined, the feedback
 *               integration is not available and feedback buttons should not
 *               be rendered.
 *
 *  Usage:
 *
 *  ```tsx
 * const openForm = useFeedbackForm();
 *
 * if (!openForm) {
 *   return null;
 * }
 *
 * return <Button onClick={() => openForm()}>{t('Give Feedback')}</button>;
```

Reply:
I like this suggestion! Does simplify it quite a bit.